src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo,
+  onTodoDelete?: (todoId: number) => void,
+  onTodoUpdate?: (todo: Todo, newTodoTitle: string) => void
+  isProcessing: boolean,
+  handleChangeTodoStatus?: (todo: Todo) => void,
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onTodoDelete = () => {},
+  onTodoUpdate = () => {},
+  isProcessing,
+  handleChangeTodoStatus = () => {},
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [todoTitle, setTodoTitle] = useState(todo.title);
+  const [isModifiedTitle, setIsModifiedTitle] = useState(false);
+
+  const handleTodoDoubleClick = () => {
+    setIsEditing(true);
+  };
+
+  const handleTodoTitleChange
+    = (event: React.ChangeEvent<HTMLInputElement>) => {
+      setTodoTitle(event.target.value);
+
+      setIsModifiedTitle(true);
+    };
+
+  const handleKeyUp = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Escape') {
+      setIsEditing(false);
+    }
+  };
+
+  const handleTodoSave = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!isModifiedTitle) {
+      setIsEditing(false);
+
+      return;
+    }
+
+    if (!todoTitle) {
+      await onTodoDelete(todo.id);
+    } else {
+      await onTodoUpdate(todo, todoTitle);
+    }
+
+    setIsEditing(false);
+  };
+
+  const titleInput = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (isEditing && titleInput.current) {
+      titleInput.current.focus();
+    }
+  }, [isEditing]);
+
+  return (
+    <div
+      key={todo.id}
+      data-cy="Todo"
+      className={cn('todo',
+        { completed: todo.completed, 'item-enter-done': true })}
```

Comment:
Why are you adding a second class with the "true" condition? It will always be executed, so this class can simply be added to the list of classes without a condition